index.js

```diff
@@ -251,6 +378,39 @@ export class Placement {
     img2.src = px.replace(/\*/, 2).replace(/\*/, random);
     beforeCheck(callback, 250)
   }
+
+  /* Returns an array of keywords (strings) found on the page
+   *
+   * @returns {Array[string]} Advertising keywords found on the page
+   */
+  detectKeywords() {
+    var keywordHist = {};  // Keywords found => count of keyword
+    const mainContent = document.querySelector("main") ||
+      document.querySelector("[role='main']") ||
+      document.querySelector("body");
+
+    const words = mainContent.textContent.split(/\s+/);
+    const wordTrimmer = /^[\('"]?(.*?)[,\.\?\!:;\)'"]?$/g
+    for (let x = 0; x < words.length && x < MAX_WORDS_ANALYZED; x++) {
+      // Remove certain punctuation from beginning and end of the word
+      let word = words[x].replace(wordTrimmer, "$1").toLowerCase();
+      if (KEYWORDS[word]) {
+        keywordHist[KEYWORDS[word]] = (keywordHist[KEYWORDS[word]] || 0) + 1;
+      }
+    }
+
+    // Sort the hist with the most common items first
+    // Grab only the MAX_KEYWORDS most common
+    const keywords = Object.entries(keywordHist).sort(
+      (a, b) => {
+        if (a[1] > b[1]) return -1;
+        if (a[1] < b[1]) return 1;
+        return 0;
+      }
+    ).slice(0, MAX_KEYWORDS).map((x) => x[0]);
```

Comment:
This is so functional it makes my brain hurt. The final `map` is basically just pulling the value from a `'keyword': count` dict item?

Reply:
Yes. `keywordHist` is a dict with keys as the keyword and the value is the count of its occurrences. `Object.entries` is basically the same as `dict.items` in Python. It returns an array of length 2 arrays (key, value). We then sort that array such that the most common items are first. Then we take the first `MAX_KEYWORDS` items from the array so as to only get the top X keywords. Lastly, we drop the number of occurrences and just return the keywords.